system/datatypes/peer_reviewed_article.php

```diff
@@ -416,14 +429,49 @@ public function editAction(HyphaRequest $request) {
 			return ['redirect', $this->constructFullPath($this->pagename)];
 		}
 
+		$formData = $request->getPostData() + $_FILES;
+
 		// create form
-		$form = $this->createEditForm($request->getPostData());
+		$form = $this->createEditForm($formData);
+
+		$featuredImageData = $form->dataFor(self::FIELD_NAME_FEATURED_IMAGE);
+		$orgImg = $this->xml->documentElement->getOrCreate(self::FIELD_NAME_FEATURED_IMAGE)->getText();
+		$hasNewImg = $featuredImageData && !$featuredImageData['error'];
+
+		if ($hasNewImg) {
+			$ext = strtolower(substr(strrchr($featuredImageData['name'], '.'), 1));
+			@$size = getimagesize($_FILES[self::FIELD_NAME_FEATURED_IMAGE]['tmp_name']);
+			if (!$size || !in_array($ext, ['jpg','jpeg','png','gif','bmp'])) {
+				$form->errors[self::FIELD_NAME_FEATURED_IMAGE] = __('invalid-image-file').ini_get('upload_max_filesize');
```

Comment:
Why append the upload max filesize, why not use the interpolation support that __ now has?

Reply:
is also like this in `loadPage()` in `pages.php` (`case HyphaRequest::HYPHA_SYSTEM_PAGE_UPLOAD`). this needs to be centralised and improved.